qp_shogun/shogun/shogun.py

```diff
@@ -183,64 +186,73 @@ def shogun(qclient, job_id, parameters, out_dir):
-            fps['raw_forward_seqs'], rs, qiime_map)
-
-        # Combining files
-        comb_fp = generate_fna_file(temp_dir, samples)
-
-        # Formatting parameters
-        parameters = _format_params(parameters, SHOGUN_PARAMS)
-
-        # Step 3 align
-        align_cmd = generate_shogun_align_commands(
-            comb_fp, temp_dir, parameters)
-        sys_msg = "Step 3 of 6: Aligning FNA with Shogun (%d/{0})".format(
-            len(align_cmd))
-        success, msg = _run_commands(
-            qclient, job_id, align_cmd, sys_msg, 'Shogun Align')
-
-        if not success:
-            return False, None, msg
-
-        # Step 4 taxonomic profile
-        sys_msg = "Step 4 of 6: Taxonomic profile with Shogun (%d/{0})"
-        assign_cmd, profile_fp = generate_shogun_assign_taxonomy_commands(
-            temp_dir, parameters)
+    rs = fps['raw_reverse_seqs'] if 'raw_reverse_seqs' in fps else []
+    samples = make_read_pairs_per_sample(
+        fps['raw_forward_seqs'], rs, qiime_map)
+
+    # Combining files
+    comb_fp = generate_fna_file(out_dir, samples)
+
+    # Formatting parameters
+    parameters = _format_params(parameters, SHOGUN_PARAMS)
+
+    # Step 3 align
+    align_cmd = generate_shogun_align_commands(
+        comb_fp, out_dir, parameters)
+    sys_msg = "Step 3 of 6: Aligning FNA with Shogun (%d/{0})".format(
+        len(align_cmd))
+    success, msg = _run_commands(
+        qclient, job_id, align_cmd, sys_msg, 'Shogun Align')
+
+    if not success:
+        return False, None, msg
+
+    # Step 4 taxonomic profile
+    sys_msg = "Step 4 of 6: Taxonomic profile with Shogun (%d/{0})"
+    assign_cmd, profile_fp = generate_shogun_assign_taxonomy_commands(
+        out_dir, parameters)
+    success, msg = _run_commands(
+        qclient, job_id, assign_cmd, sys_msg, 'Shogun taxonomy assignment')
+    if not success:
+        return False, None, msg
+
+    sys_msg = "Step 5 of 6: Compressing and converting alignment to BIOM"
+    qclient.update_job_step(job_id, msg)
+    alignment_fp = join(out_dir, 'alignment.%s.%s' % (
+        parameters['aligner'], ALN2EXT[parameters['aligner']]))
+    xz_cmd = 'xz -9 -T%s %s' % (parameters['threads'], alignment_fp)
+    std_out, std_err, return_value = system_call(xz_cmd)
+    if return_value != 0:
+        error_msg = ("Error during %s:\nStd out: %s\nStd err: %s"
+                     "\n\nCommand run was:\n%s"
+                     % (sys_msg, std_out, std_err, xz_cmd))
+        return False, None, error_msg
+    output = run_shogun_to_biom(profile_fp, [None, None, None, True],
+                                out_dir, 'profile')
+
+    ainfo = [ArtifactInfo('Shogun Alignment Profile', 'BIOM',
+                          [(output, 'biom'),
+                           ('%s.xz' % alignment_fp, 'log')])]
+
+    # Step 5 redistribute profile
+    sys_msg = "Step 6 of 6: Redistributed profile with Shogun (%d/{0})"
+    levels = ['phylum', 'genus', 'species']
```

Comment:
I see that you kept the original design of three levels. I think this is okay. In the future we can consider expanding to all eight levels. Some users may love to see "strain-level" resolution.

Reply:
I can add, should we do 4 levels (5 with alignment: ['phylum', 'genus', 'species', 'strain']) or should we do 3 levels (4 with alignment: ['genus', 'species', 'strain'])?

Reply:
How about doing 'all', which will calculate all eight levels? I am bit hesitated to specify the individual level 'strain', because 'strain' is something very tricky (it is not an official NCBI level, but something artificial to describe below-species levels).

Reply:
All will be too many artifacts and can get confusing for users; thus limiting to 4-5 will be best; for example: 
<img width="180" alt="Screen Shot 2020-01-09 at 2 46 18 PM" src="https://user-images.githubusercontent.com/2014559/72107573-d5df2e00-32ee-11ea-8040-0971b85c5502.png">

Reply:
I see your point! How about we just keep the original three (phylum, genus, species)? I think none of the three databases can actually do strain-level resolution right.

Reply:
Sounds good, which is what is already implemented. Let's just wait for the test cause I forgot to update the version of the plugin.